File: radionum.js
const { performance } = require('perf_hooks');

function isValid(n) {
  d = null;
  c = false;

  while (n > 0) {
    if ((d != null) && (d == (n % 10))) {
      if (c) {
        return false;
      }
      c = true;
    } else {
      c = false;
    }

    d = n % 10;
    n = Math.trunc(n / 10);
  }

  return true;
}

function generateValidNumberList() {
  numberList = [];

  for (var i = 10; i < 99; i++) {
    if (isValid(i)) {
      numberList.push(i);
    }
  }

  for (var i = 1000; i < 9999; i++) {
    if (isValid(i)) {
      numberList.push(i);
    }
  }

  for (var i = 100000; i < 999999; i++) {
    if (isValid(i)) {
      numberList.push(i);
    }
  }

  return numberList;
}

function format(l) {
}

var t0 = performance.now();
generateValidNumberList();
var t1 = performance.now();
var dt = t1 - t0;

console.log(dt);

File: radionum.go
package main

import (
	"fmt"
	//	"io/ioutil"
	"strconv"
	"time"
)

func isValid(n int) bool {
	var d int
	dv := false
	c := false

	for n > 0 {
		if dv && (d == (n % 10)) {
			if c {
				return false
			}
			c = true
		} else {
			dv = true
			c = false
		}

		d = n % 10
		n = n / 10
	}

	return true
}

func GenerateValidNumberList() []int {
	var l []int

	for i := 10; i <= 99; i++ {
		if isValid(i) {
			l = append(l, i)
		}
	}

	for i := 1000; i <= 9999; i++ {
		if isValid(i) {
			l = append(l, i)
		}
	}

	for i := 100000; i <= 999999; i++ {
		if isValid(i) {
			l = append(l, i)
		}
	}

	return l
}

func format(l []int) []byte {
	var fl []byte

	for _, e := range l {
		s := strconv.Itoa(e) + "\n"
		fl = append(fl, []byte(s)...)
	}

	return fl
}

func main() {
	t0 := time.Now()

	_ = GenerateValidNumberList()
	/*
		s := format(l)
		if err := ioutil.WriteFile("valid_station_numbers-go.txt", s, 0644); err != nil {
			panic(err)
		}
	*/

	dt := time.Since(t0)
	fmt.Println(dt)
}
